### ISSUE TITLE
broken link to lifecycle hooks

### ISSUE THREAD

**Issue by manuks** [angular/angular#53794]
### Describe the problem that you experienced

It shows "page not found" on the lifecycle hooks link. Instead of using the link https://angular.dev/guide/components/lifecycle it is using the link https://angular.dev/guide/lifecycle-hooks

### Enter the URL of the topic with the problem

https://angular.dev/tutorials/learn-angular

### Add a screenshot if that helps illustrate the problem

![image](https://github.com/angular/angular/assets/570393/c42b1d08-835e-4212-a393-d001968b330b)
The link to the `Lifecycle Hooks` in the popup above `@Component` decorator is pointing to https://angular.dev/guide/lifecycle-hooks which shows as below
![image](https://github.com/angular/angular/assets/570393/686431bd-17d4-401a-8956-0efeeff6214b)

### Provide any additional information here in as much as detail as you can

```true
The link should be https://angular.dev/guide/components/lifecycle
```

**Comment by JeanMeche** [comment]
Hi, this link is currently relative in the docs and matches the page for angular.io. 

https://github.com/angular/angular/blob/1c63eddb7cc22df772e47776afb491abe963aeba/packages/core/src/metadata/directives.ts#L403

https://github.com/angular/angular/blob/main/packages/core/src/metadata/directives.ts#L403

We'll fix it once we deprecate anguiar.io and make angular.dev stable.

**Comment by JeanMeche** [comment]
It will be fixed by #55043
